tests/test_ec2/test_iam_instance_profile_associations.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import unicode_literals
+
+# Ensure 'pytest.raises' context manager support for Python 2.6
+import pytest
+
+import boto3
+from botocore.exceptions import ClientError
+import sure  # noqa
+
+from moto import mock_ec2, mock_iam
+
+
+@mock_ec2
+@mock_iam
+def test_iam_instance_profile_associations():
```

Comment:
Can this test be split up? I realize that there will be some overhead, but ideally each test method has a singular purpose. (test_associate, test_describe, test_disassociate, test_invalid_disassociate, etc)

Reply:
Done